### PR TITLE
Refactor cli to use yargs

### LINKED ISSUE #331: cli: doc: OPTIONS output not working
When I run the `--help` command in the CLI for js-ipfs commands, I get the following: 

```
$ node src/cli/bin.js id --help
Usage: bin.js id [OPTIONS]

  Shows IPFS Node ID info
```

What does `[OPTIONS]` mean? Shouldn't it show the options? Let's look at id.js: 

```js
module.exports = Command.extend({
  desc: 'Shows IPFS Node ID infos',

  options: {
    format: {
      alias: 'f',
      type: 'string'
    }
  },

  run: (name) => {
    ...
  }
})
```

Why isn't `[--format | -f]` output?

### LINKED ISSUE #377: `jsipfs version` should work without any ipfs repo
I am adding sharness tests to jsipfs and it fails very early because `jsipfs` expects an ipfs repo:

```
$ IPFS_PATH='stuff' jsipfs version
error   no ipfs repo found
$ IPFS_PATH='stuff' ipfs version
ipfs version 0.4.2
```

Also it looks like jsipfs error messages are printed on stdout instead of stderr.

### PR DESCRIPTION
**Pull request by dignifiedquire**
Make the cli interface a little bit more awesome :)

- [x] Replace `ronin` with `yargs`, closes #331
- [x] adds an auto update check to npm once a week to see if your version is the latest one :)
- [x] Update `version` command to return the actual version of `js-ipfs` not the version in the config. Closes #377

### CODE REVIEW

**Comment by RichardLitt** [comment]
Subscribing!

**Comment by RichardLitt** [comment]
Why not minimist?

**Comment by dignifiedquire** [comment]
Because minimist is similarly to ronin not really maintained anymore at the moment.

**Comment by RichardLitt** [comment]
@dignifiedquire Can we talk to someone about that? Maybe @substack would be interested in handing it over to someone, or something?

**Comment by dignifiedquire** [comment]
You can but minimist ist nowhere near what we need in terms of functionality, and yargs being the successor of optimist is a very well developers

**Comment by dignifiedquire** [comment]
This is now ready for review @RichardLitt @diasdavid 

All failing tests are as far as I can see from the duplexify issues, but will wait for CI to confirm that.

File: src/cli/bin.js [hunk 1]
```diff
@@ -2,12 +2,20 @@
...
+const yargs = require('yargs')
+const updateNotifier = require('update-notifier')
+const readPkgUp = require('read-pkg-up')
 
-const cli = ronin(__dirname)
+const pkg = readPkgUp.sync().pkg
+updateNotifier({
+  pkg,
+  updateCheckInterval: 1000 * 60 * 60 * 24 * 7 // 1 week
+}).notify()
```
Comment (diasdavid) [inline src/cli/bin.js]: 👍

File: src/cli/commands/bitswap.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+'use strict'
+
+module.exports = {
+  command: 'bitswap',
+
+  description: 'A set of commands to manipulate the bitswap agent.',
+
+  builder (yargs) {
+    return yargs
+      .commandDir('bitswap')
```
Comment (diasdavid) [inline src/cli/commands/bitswap.js]: Is building the subcommands inside the main command a best practice? Didn't see it on the module description.

**Comment by diasdavid** [comment]
Does this solve the 'help text' issue for the commands?

File: src/cli/commands/bitswap.js (hunk 2 above)
Comment (dignifiedquire) [inline src/cli/commands/bitswap.js]: Yes this this is the recommended (and as far as I can tell only way) to do this.

**Comment by dignifiedquire** [comment]
> Does this solve the 'help text' issue for the commands?

yes

**Comment by dignifiedquire** [comment]
@diasdavid getting green, please review :)

**Comment by dignifiedquire** [comment]
New output

```bash
node src/cli/bin.js files --help
src/cli/bin.js files

Commands:
  add <file>       Add a file to IPFS using the UnixFS
                   data format
  cat <ipfs-path>  Download IPFS objects
  get <ipfs-path>  Download IPFS objects

Options:
  --help  Show help                           [boolean]


js-ipfs git/better-cli
❯ node src/cli/bin.js files add --help
src/cli/bin.js files add <file>

Options:
  --help           Show help                  [boolean]
  --recursive, -r            [boolean] [default: false]
```

**Comment by diasdavid** [comment]
nice, just tried it out, I like it :) let's merge it it :)

**Comment by diasdavid** [comment]
@RichardLitt how is the ipfs mans coming along?

**Comment by RichardLitt** [comment]
I haven't had much bandwidth for this since getting back. :(

**Pull request merged**